tests/test_authority.py

```diff
@@ -123,3 +128,104 @@ class MockResponse(object):
+    "token_endpoint": "https://contoso.com/placeholder",
+    }
+
+@patch("msal.authority.tenant_discovery", return_value=_mocked_tenant_discovery_response)
+@patch("msal.authority.instance_discovery")
+@patch.object(msal.PublicClientApplication, "_get_instance_metadata", return_value=[])
+class TestMsalAuthorityBehaviorsWithoutKnownToDeveloperAuthorities(unittest.TestCase):
+
+    def test_a_known_to_microsoft_authority_should_skip_validation_but_still_use_instance_metadata(
+            self, instance_metadata, known_to_microsoft_validation, _):
+        app = msal.PublicClientApplication(
+            "id", authority="https://login.microsoftonline.com/common")
+        known_to_microsoft_validation.assert_not_called()
+        app.get_accounts()  # This could make an instance metadata call for authority aliases
+        instance_metadata.assert_called_once_with()
+
+    def test_validate_authority_boolean_should_skip_validation_and_instance_metadata(
+            self, instance_metadata, known_to_microsoft_validation, _):
+        """Pending deprecation, but kept for backward compatibility, for now"""
+        app = msal.PublicClientApplication(
+            "id", authority="https://contoso.com/common", validate_authority=False)
+        known_to_microsoft_validation.assert_not_called()
+        app.get_accounts()  # This could make an instance metadata call for authority aliases
+        instance_metadata.assert_not_called()
+
+    def test_by_default_adfs_should_skip_validation_and_instance_metadata(
+            self, instance_metadata, known_to_microsoft_validation, _):
+        """Not strictly required, but when/if we already supported it, we better keep it"""
+        app = msal.PublicClientApplication("id", authority="https://contoso.com/adfs")
+        known_to_microsoft_validation.assert_not_called()
+        app.get_accounts()  # This could make an instance metadata call for authority aliases
+        instance_metadata.assert_not_called()
+
+    def test_by_default_b2c_should_skip_validation_and_instance_metadata(
+            self, instance_metadata, known_to_microsoft_validation, _):
+        """Not strictly required, but when/if we already supported it, we better keep it"""
+        app = msal.PublicClientApplication(
+            "id", authority="https://login.b2clogin.com/contoso/b2c_policy")
+        known_to_microsoft_validation.assert_not_called()
+        app.get_accounts()  # This could make an instance metadata call for authority aliases
+        instance_metadata.assert_not_called()
+
+@patch("msal.authority.tenant_discovery", return_value=_mocked_tenant_discovery_response)
+@patch("msal.authority.instance_discovery")
+@patch.object(msal.PublicClientApplication, "_get_instance_metadata", return_value=[])
+class TestMsalAuthorityBehaviorsWithKnownToDeveloperAuthorities(unittest.TestCase):
+
+    def test_known_authorities_should_allow_multiple_configuration_but_only_one_modification(self, *mocks):
```

Comment:
What does 'multiple configuation' mean here? Also, why are we allowing 1 modification?

Reply:
Good catch! Sorry for ambiguity. Let me change it in next commit. I mean to say "multiple idempotent set operation but zero modification".